### ISSUE TITLE
CoreML: Resolution Time Regressor INTEGRATION

### ISSUE THREAD

**Issue by ALearningCurve** [ALearningCurve/ticket-forge#27]
Develop a lightweight supervised learning model to provide the "Complexity" aspect of the triage via estimating the completion time in hours.

Previous tickets have prepared the data and the ML training framework:
- #10 
- #4 

This ticket requires integration with #22 and changing the logic from #10 to use the actual data/optimize models

Some requirements for the training:
- Code needs to live in https://github.com/ALearningCurve/ticket-forge/tree/main/apps/training
- Switch current approach which has fixed train/test/validation splits to train/test w/k-fold if there is small dataset
- Add more models, experiment with different hyper-parameters and feature engineering, and so on to maximize accuracy
- Add tests which check for regresssions in training process (i.e. accuracy, time, etc.)

Suggestions:
- Results should be stored using DVC or some kind of experiment tracking system (#22)
- Use only sklearn models and xgboost's sklearn api for now since POC
    - If you want to get fancy with it, add some NN (i.e. simple FNN, CNN, longformer, etc.) 
- Seed everything

---

Scenario 1: we are able to predict the complexity of tickets
**Given**: Historical ticket data including embeddings, labels, and actual time-to-close.
**When**: A new ticket is ingested into the regression model.
**Then**: The system outputs an estimated resolution time in hours with a Mean Absolute Error (MAE) target of less than 8 hours.
